controllers/webhook.js

```diff
@@ -0,0 +1,260 @@
+'use strict'
+
+const bufferEq = require('buffer-equal-constant-time')
+const path = require('path')
+const config = require(path.join(__dirname, '/../config'))
+const crypto = require('crypto')
+
+const gitFactory = require('../lib/GitServiceFactory')
+const Staticman = require('../lib/Staticman')
+
+/**
+ * Express handler for webhook requests/notifications generated by the backing git service.
+ */
+module.exports = async (req, res, next) => {
+  // Allow for multiple errors to be raised and reported back.
+  let errorsRaised = []
+  let event
+
+  let service = req.params.service
+  const version = req.params.version
+  // v1 of the webhook endpoint assumed GitHub.
+  if (service === null || typeof service === 'undefined') {
+    if (version === '1') {
+      service = 'github'
+    }
+  }
```

Comment:
I could be wrong, but I think this could safely be changed to the following

```suggestion
  if (!service && version === '1') {
    service = 'github'
  }
```

I removed the comment here as well. It's one that's useful in code review but I don't think we need it in the codebase.

Reply:
Changed as requested. In general, I've always felt a little uneasy relying on truthiness and falsiness. But, again, terseness. :)